BorrowBookUI.java

```diff
@@ -0,0 +1,128 @@
+import java.util.Scanner;
+
+
+public class BorrowBookUI {
+	
+	public static enum UIState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
+
+	private BorrowBookControl control;
+	private Scanner input;
+	private UIState state;
+
+	
+	public BorrowBookUI(BorrowBookControl control) {
+		this.control = control;
+		input = new Scanner(System.in);
+		this.state = UIState.INITIALISED;
+		this.control.setUI(this);
+		/*control.setUI(this);*/
+	}
+
+	
+	private String input(String prompt) {
+		/*System.out.print(prompt);*/
+        System.out.print(this);
+		return input.nextLine();
+	}	
+		
+		
+	private void output(Object object) {
+
+    	System.out.println(object);
+	}
+	
+			
+	public void setState(UIState STATE) {
+
+		this.state = STATE;
+
+	}
+
+	
+	public void run() {
+
+		output("Borrow Book Use Case UI\n");
+		
+		while (true) {
+			
+			switch (state) {
+			
+			case CANCELLED:
+				output("Borrowing Cancelled");
```

Comment:
DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier